src/components/ProfileCard/BasicDetailsEdit/hooks.ts

```diff
@@ -64,7 +71,46 @@ const useUpdateProfileDetails = () => {
     }
   };
 
+
+
   return updateProfileWorkplace;
 };
 
-export default useUpdateProfileDetails;
+interface IReturnType {
+  data: any;
+  isLoading: boolean;
+  refetch: () => any;
+}
+
+const fetchNFTs = (walletId: any, agree: boolean): IReturnType => {
+  const chain = 'polygon';
+  const { data, isLoading, refetch } = useQuery(
+    'fetchNFTs',
+    async () => {
+      const response = await fetch(
+        `https://deep-index.moralis.io/api/v2/${walletId}/nft?chain=${chain}`,
+        {
+          method: 'GET',
+          headers: {
+            'Content-Type': 'application/json',
+            'X-API-Key': import.meta.env.VITE_MORALIS_API_KEY || '',
+          },
+        }
+      );
+      const res = await response.json();
+      return res;
+    },
+    {
+      retry: false,
+      staleTime: Infinity,
+      refetchOnWindowFocus: false,
+      onError: (error: any) => {
+        handleError({ error, explicitMessage: 'Unable to fetch nfts' });
+      },
+      enabled: false,
```

Comment:
What was the use of enabled?

Reply:
This stops the fetchNFTs query to run automatically, I am running it manually later when end-user accepts to share his NFT data.